ex4.py

```diff
@@ -0,0 +1,17 @@
+import numpy as np
+
+
+def add_vector(matrix, vector):
+    """
+    Adds a given vector to each row of a given matrix.
+    :param matrix: The original matrix.
+    :param vector: The vector to add to each row.
+    :return: The altered matrix.
+    """
+    return np.asarray(matrix) + np.asarray(vector)
```

Comment:
Why are you converting the matrix and vector?  If you would use type hinting this would be clearer.